src/libostree/ostree-repo-pull.c

```diff
@@ -381,12 +388,80 @@ static void
 enqueue_one_object_request (OtPullData        *pull_data,
                             const char        *checksum,
                             OstreeObjectType   objtype,
+                            const char        *path,
                             gboolean           is_detached_meta,
                             gboolean           object_is_stored);
 
 static gboolean
+matches_pull_dir (const char *current_file,
```

Comment:
If it was hard to write it should be hard to read...

Honestly, I don't know why things like this are so tricky. I wonder if there is a way to formulate it that is better.